components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolverTest.java

```diff
@@ -0,0 +1,254 @@
+
+        assertEquals(primaryKey, identityKeyStoreResolver.getPrivateKey(SUPER_TENANT_DOMAIN, InboundProtocol.WS_FEDERATION));
+        assertEquals(tenantKey, identityKeyStoreResolver.getPrivateKey(TENANT_DOMAIN, InboundProtocol.WS_FEDERATION));
+        assertEquals(customKey, identityKeyStoreResolver.getPrivateKey(SUPER_TENANT_DOMAIN, InboundProtocol.OAUTH));
+        assertEquals(customKey, identityKeyStoreResolver.getPrivateKey(TENANT_DOMAIN, InboundProtocol.OAUTH));
+        assertEquals(customKey, identityKeyStoreResolver.getPrivateKey(SUPER_TENANT_DOMAIN, InboundProtocol.WS_TRUST));
+        assertEquals(tenantKey, identityKeyStoreResolver.getPrivateKey(TENANT_DOMAIN, InboundProtocol.WS_TRUST));
+    }
+
+    @Test
+    public void testGetCertificate() throws Exception {
+
+        System.setProperty(CarbonBaseConstants.CARBON_HOME,
+                Paths.get(System.getProperty("user.dir"), "src", "test", "resources").toString());
+
+        Certificate primaryCertificate = getKeyStoreFromFile(PRIMARY_KEY_STORE, PRIMARY_KEY_STORE_PASSWORD,
+                System.getProperty(CarbonBaseConstants.CARBON_HOME)).getCertificate(PRIMARY_KEY_STORE_ALIAS);
+        Certificate tenantCertificate = getKeyStoreFromFile(TENANT_KEY_STORE, TENANT_KEY_STORE_PASSWORD,
+                System.getProperty(CarbonBaseConstants.CARBON_HOME)).getCertificate(TENANT_KEY_STORE_ALIAS);
+        Certificate customCertificate = getKeyStoreFromFile(CUSTOM_KEY_STORE, CUSTOM_KEY_STORE_PASSWORD,
+                System.getProperty(CarbonBaseConstants.CARBON_HOME)).getCertificate(CUSTOM_KEY_STORE_ALIAS);
+
+        Map<String, Certificate> publicCerts = new ConcurrentHashMap<>();
+        publicCerts.put(SUPER_TENANT_ID, primaryCertificate);
+        publicCerts.put(TENANT_ID, tenantCertificate);
+        publicCerts.put(InboundProtocol.OAUTH.toString(), customCertificate);
+        publicCerts.put(InboundProtocol.WS_TRUST.toString(), customCertificate);
+
+        setFinalStatic(IdentityKeyStoreResolver.class.getDeclaredField("publicCerts"), publicCerts);
+
+        assertEquals(primaryCertificate, identityKeyStoreResolver.getCertificate(SUPER_TENANT_DOMAIN, InboundProtocol.WS_FEDERATION));
+        assertEquals(tenantCertificate, identityKeyStoreResolver.getCertificate(TENANT_DOMAIN, InboundProtocol.WS_FEDERATION));
+        assertEquals(customCertificate, identityKeyStoreResolver.getCertificate(SUPER_TENANT_DOMAIN, InboundProtocol.OAUTH));
+        assertEquals(customCertificate, identityKeyStoreResolver.getCertificate(TENANT_DOMAIN, InboundProtocol.OAUTH));
+        assertEquals(customCertificate, identityKeyStoreResolver.getCertificate(SUPER_TENANT_DOMAIN, InboundProtocol.WS_TRUST));
+        assertEquals(tenantCertificate, identityKeyStoreResolver.getCertificate(TENANT_DOMAIN, InboundProtocol.WS_TRUST));
+    }
+
+    private void setFinalStatic(Field field, Object newValue) throws Exception {
```

Comment:
Can you please explain the use of this method

Reply:
Improved by moving the logic to `setPrivateStaticField` method